doc/modules/calibration.rst

```diff
@@ -205,3 +205,124 @@ a similar decrease in log-loss.
     .. [5] On the combination of forecast probabilities for
            consecutive precipitation periods. Wea. Forecasting, 5, 640–650.,
            Wilks, D. S., 1990a
+
+
+.. _decision_threshold_calibration:
+
+==============================
+Decision Threshold calibration
+==============================
+
+.. currentmodule:: sklearn.calibration
+
+Often Machine Learning classifiers base their
+predictions on real-valued decision functions or probability estimates that
+carry the inherited biases of their models. Additionally when using a machine
+learning model the evaluation criteria can differ from the optimisation
+objectives used by the model during training.
+
+When predicting between two classes it is commonly advised that an appropriate
+decision threshold is estimated based on some cutoff criteria rather than
+arbitrarily using the midpoint of the space of possible values. Estimating a
+decision threshold for a specific use case can help to increase the overall
+accuracy of the model and provide better handling for sensitive classes.
+
+.. currentmodule:: sklearn.calibration
+
+:class:`CutoffClassifier` can be used as a wrapper around a model for binary
+classification to help obtain a more appropriate decision threshold and use it
+for predicting new samples.
+
+Usage
+-----
+
+To use the :class:`CutoffClassifier` you need to provide an estimator that has
+a ``decision_function`` or a ``predict_proba`` method. The ``method``
+parameter controls whether the first will be preferred over the second if both
+are available.
+
+The wrapped estimator can be pre-trained, in which case ``cv = 'prefit'``, or
+not. If the classifier is not trained then a cross-validation loop specified by
+the parameter ``cv`` can be used to obtain a decision threshold by averaging
+all decision thresholds calculated on the hold-out parts of each cross
+validation iteration. Finally the model is trained on all the provided data.
+When using ``cv = 'prefit'`` you need to make sure to use a hold-out part of
+your data for calibration.
+
+The strategies, controlled by the parameter ``strategy``, for finding
+appropriate decision thresholds are based either on precision recall estimates
+or true positive and true negative rates. Specifically:
+
+.. currentmodule:: sklearn.metrics
+
+* ``f_beta``
+   selects a decision threshold that maximizes the :func:`fbeta_score`. The
+   value of beta is specified by the parameter ``beta``. The ``beta`` parameter
+   determines the weight of precision. When ``beta = 1`` both precision recall
+   get the same weight therefore the maximization target in this case is the
+   :func:`f1_score`. if ``beta < 1`` more weight is given to precision whereas
+   if ``beta > 1`` more weight is given to recall.
+
+* ``roc``
+   selects the decision threshold for the point on the :func:`roc_curve` that
+   is closest to the ideal corner (0, 1)
+
+* ``max_tpr``
```

Comment:
We might want to think about how to avoid confusion with the `roc_auc_score` parameter, `max_fpr` which means "only get the area under a section of the curve below a specified maximum false positive rate", whereas here, `max_*` means "maximise * while constraining its counterpart"

Reply:
hmm.. confusing indeed, what if we renamed it to `maximise_tpr`?

Reply:
Yes, `maximize_tpr`, or just `tpr`?